components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/selection/GeckoSelectionActionDelegate.kt

```diff
@@ -0,0 +1,59 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.engine.gecko.selection
+
+import android.app.Activity
+import android.content.Context
+import android.view.MenuItem
+import mozilla.components.concept.engine.selection.SelectionActionDelegate
+import org.mozilla.geckoview.BasicSelectionActionDelegate
+
+/**
+ * An adapter between the GV [BasicSelectionActionDelegate] and a generic [SelectionActionDelegate].
+ *
+ * @param customDelegate handles as much of this logic as possible.
+ */
+class GeckoSelectionActionDelegate(
+    activity: Activity,
+    private val customDelegate: SelectionActionDelegate
+) : BasicSelectionActionDelegate(activity) {
+
+    companion object {
+        /**
+         * @returns a [GeckoSelectionActionDelegate] if [customDelegate] is non-null and [context]
+         * is an instance of [Activity]. Otherwise, returns null.
+         */
+        fun maybeCreate(context: Context, customDelegate: SelectionActionDelegate?): GeckoSelectionActionDelegate? {
+            return if (context !is Activity || customDelegate == null) {
```

Comment:
Personally I think this is easier to read with the truthy statement first
```suggestion
            return if (context is Activity && customDelegate != null) {
```